src/main/java/de/tum/in/www1/artemis/service/ComplaintService.java

```diff
@@ -127,16 +138,54 @@ public long countComplaintsByExerciseId(long exerciseId) {
         }
 
         databaseComplaints.get().forEach(complaint -> {
-            String submissorName = principal.getName();
-            User assessor = complaint.getResult().getAssessor();
+            if (complaint.getComplaintType() != ComplaintType.MORE_FEEDBACK) {
```

Comment:
You should not change data in the Service just so that the client does not see it, because this can lead to problems: the transaction mechanism might automatically change the values in the database. Therefore we have one very important rule and follow the MVC pattern:

1) Model: all domain + repository classes for data retrieval and storage
2) Controller: all service classes. Use `Transactional` annotation and should NEVER hide information to the client.
3) View: all resources classes. Manage request validation (e.g. access handling) and information hiding (e.g. assessor name for students). Should NEVER have `Transactional` annotation.

@maxr96 please collect this for some server coding and consistency guidelines. I would ask you to also check this in other places of the application and create JIRA bugs in case you notice deviations from these rules. Thank you!